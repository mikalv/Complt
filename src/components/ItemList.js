import React from 'react';
import { List } from 'material-ui/List';
import Item from './Item';
import OakPropTypes from '../PropTypes';

const ItemList = ({ items, onItemAvatarTap }) => (
  <List>
    {items.map((item, i) =>
      <Item id={i} key={i} item={item} onAvatarTouchTap={() => onItemAvatarTap(i)} />)}
  </List>
);

ItemList.propTypes = {
  items: React.PropTypes.arrayOf(OakPropTypes.item),
  onItemAvatarTap: React.PropTypes.func,
};

export default ItemList;
