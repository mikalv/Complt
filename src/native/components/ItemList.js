import React from 'react';
import { ScrollView } from 'react-native';
import PropTypes from '../../common/PropTypes';
import Item from './Item';


const List = props => (
  <ScrollView>
    {props.items.map((item, i) => {
      if (!item) return null;
      return (<Item
        key={item._id}
        item={item}
        onAvatarPress={props.onAvatarPress !== undefined ? () => props.onAvatarPress(i) : undefined}
        onItemPress={props.onAvatarPress !== undefined ? () => props.onAvatarPress(i) : undefined}
        onDeletePress={props.onDeletePress !== undefined ? () => props.onDeletePress(i) : undefined}
      />);
    })}
  </ScrollView>
);

List.propTypes = {
  items: React.PropTypes.arrayOf(PropTypes.item),
  onAvatarPress: React.PropTypes.func, // eslint-disable-line react/no-unused-prop-types
  onItemPress: React.PropTypes.func, // eslint-disable-line react/no-unused-prop-types
  onDeletePress: React.PropTypes.func,
};

export default List;
