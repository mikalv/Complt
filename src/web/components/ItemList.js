import React from 'react';
import Item from './Item';
import PropTypes from '../../common/PropTypes';

const ItemList = ({
    items = [],
    onItemAvatarTap,
    style,
    onDelete,
    canDeleteTask,
    canDeleteProject,
    canMove,
    onItemTap,
    onItemUpdate,
    onItemMove,
    className,
}) => (
  <div style={style} className={className}>
    {items.map((item, i) => {
      if (!item) return null;
      return (<Item
        key={item._id}
        item={item}
        canDelete={
          (canDeleteTask && item.isProject === false) ||
          (canDeleteProject && item.isProject === true)}
        canMove={canMove}
        onAvatarTouchTap={onItemAvatarTap !== undefined ? () => onItemAvatarTap(i) : undefined}
        onItemTap={onItemTap !== undefined ? () => onItemTap(i) : undefined}
        onItemUpdate={onItemUpdate !== undefined ? () => onItemUpdate(i) : undefined}
        onItemMove={onItemMove !== undefined ? () => onItemMove(i) : undefined}
        onDelete={() => onDelete(i)}
      />);
    })}
  </div>
);

ItemList.propTypes = {
  items: React.PropTypes.arrayOf(PropTypes.item),
  onItemAvatarTap: React.PropTypes.func,
  canDeleteProject: React.PropTypes.bool,
  canDeleteTask: React.PropTypes.bool,
  canMove: React.PropTypes.bool,
  onDelete: React.PropTypes.func,
  onItemTap: React.PropTypes.func,
  onItemUpdate: React.PropTypes.func,
  onItemMove: React.PropTypes.func,
  style: React.PropTypes.object, // eslint-disable-line react/forbid-prop-types
  className: React.PropTypes.string,
};

export default ItemList;
