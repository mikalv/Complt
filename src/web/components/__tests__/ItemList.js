import React from 'react';
import { mount } from 'enzyme';
import ItemList from '../ItemList';

const task = { name: 'Task', isProject: false, isCompleted: false };
const project = { name: 'Project', isProject: true };

describe('ItemList component', () => {
  it('renders without crashing with no items', () => {
    mount(<ItemList />);
  });
  it('renders without crashing when an item is null', () => {
    mount(<ItemList items={[null]} />);
  });
  it('renders an item correctly', () => {
    const component = mount(<ItemList canMove items={[task]} />);
    expect(component).toMatchSnapshot();
  });
  it('calls onDelete correctly with a task', () => {
    const onDelete = jest.fn();
    const component = mount(<ItemList
      canDeleteTask
      onDelete={onDelete}
      items={[project, task, project]}
    />);
    component.childAt(2).find('MdDelete').simulate('click');
    expect(onDelete).toBeCalledWith(1);
  });
  it('calls onDelete correctly with a project', () => {
    const onDelete = jest.fn();
    const component = mount(<ItemList
      canDeleteProject
      onDelete={onDelete}
      items={[project, task, project]}
    />);
    component.childAt(4).find('MdDelete').simulate('click');
    expect(onDelete).toBeCalledWith(2);
  });
  it('calls onItemTap correctly', () => {
    const onItemTap = jest.fn();
    const component = mount(<ItemList
      onItemTap={onItemTap}
      items={[project, task, project]}
    />);
    component.childAt(4).find('.Item-center').simulate('click');
    expect(onItemTap).toBeCalledWith(2);
  });
  it('calls onLeftButtonClick correctly', () => {
    const onLeftButtonClick = jest.fn();
    const component = mount(<ItemList
      onLeftButtonClick={onLeftButtonClick}
      items={[project, task, project]}
    />);
    component.childAt(4).find('.Item-left').find('button').simulate('click');
    expect(onLeftButtonClick).toBeCalledWith(2);
  });
  it('calls onItemUpdate correctly', () => {
    const onItemUpdate = jest.fn();
    const component = mount(<ItemList
      onItemUpdate={onItemUpdate}
      items={[project, task, project]}
    />);
    component.childAt(4).find('MdCreate').simulate('click');
    expect(onItemUpdate).toBeCalledWith(2);
  });
  it('calls onItemMove correctly', () => {
    const onItemMove = jest.fn();
    const component = mount(<ItemList
      onItemMove={onItemMove}
      canMove
      items={[project, task, project]}
    />);
    component.childAt(4).find('MdMoreVert').simulate('click');
    expect(onItemMove).toBeCalledWith(2);
  });
});
