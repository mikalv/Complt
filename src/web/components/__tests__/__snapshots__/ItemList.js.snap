exports[`ItemList component renders an item correctly 1`] = `
<Component
  canMove={true}
  items={
    Array [
      Object {
        "isCompleted": false,
        "isProject": false,
        "name": "Task",
      },
    ]
  }>
  <div>
    <Component
      canMove={true}
      index={0}
      item={
        Object {
          "isCompleted": false,
          "isProject": false,
          "name": "Task",
        }
      }
      onDelete={[Function]}>
      <AccessibleFakeButtonInked
        className="flex row Item md-text"
        component="div"
        inkTransitionEnterTimeout={450}
        inkTransitionLeaveTimeout={300}
        inkTransitionOverlap={150}>
        <AccessibleFakeButton
          className="flex row Item md-text"
          component="div"
          ink={
            <InkContainer
              className={undefined}
              disabledInteractions={undefined}
              inkClassName={undefined}
              inkStyle={undefined}
              style={undefined}
              transitionEnterTimeout={450}
              transitionLeaveTimeout={300}
              transitionOverlap={150}
              waitForInkTransition={undefined} />
          }
          role="button"
          tabIndex={0}>
          <div
            aria-pressed={false}
            className="md-fake-btn md-pointer--hover flex row Item md-text"
            onBlur={[Function]}
            onClick={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            role="button"
            tabIndex={0}>
            <InkContainer
              transitionEnterTimeout={450}
              transitionLeaveTimeout={300}
              transitionOverlap={150}>
              <ReactTransitionGroup
                childFactory={[Function]}
                className="md-ink-container"
                component="div">
                <div
                  className="md-ink-container" />
              </ReactTransitionGroup>
            </InkContainer>
            <div
              className="Item-left">
              <ButtonTooltipedInked
                icon={true}
                inkTransitionEnterTimeout={450}
                inkTransitionLeaveTimeout={300}
                inkTransitionOverlap={150}>
                <ButtonTooltiped
                  icon={true}
                  ink={
                    <InkContainer
                      className={undefined}
                      disabledInteractions={undefined}
                      inkClassName={undefined}
                      inkStyle={undefined}
                      style={undefined}
                      transitionEnterTimeout={450}
                      transitionLeaveTimeout={300}
                      transitionOverlap={150}
                      waitForInkTransition={undefined} />
                  }
                  tooltipDelay={0}
                  tooltipPosition="bottom"
                  tooltipTransitionEnterTimeout={150}
                  tooltipTransitionLeaveTimeout={150}>
                  <Button
                    fixedPosition="br"
                    icon={true}
                    iconBefore={true}
                    ink={
                      <InkContainer
                        className={undefined}
                        disabledInteractions={undefined}
                        inkClassName={undefined}
                        inkStyle={undefined}
                        style={undefined}
                        transitionEnterTimeout={450}
                        transitionLeaveTimeout={300}
                        transitionOverlap={150}
                        waitForInkTransition={undefined} />
                    }
                    type="button">
                    <button
                      className="md-inline-block md-btn md-btn--icon md-pointer--hover"
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseOver={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchStart={[Function]}
                      type="button">
                      <InkContainer
                        transitionEnterTimeout={450}
                        transitionLeaveTimeout={300}
                        transitionOverlap={150}>
                        <ReactTransitionGroup
                          childFactory={[Function]}
                          className="md-ink-container"
                          component="div">
                          <div
                            className="md-ink-container" />
                        </ReactTransitionGroup>
                      </InkContainer>
                      <FontIcon
                        iconClassName="material-icons">
                        <i
                          className="md-icon material-icons">
                          <MdDone>
                            <IconBase
                              viewBox="0 0 40 40">
                              <svg
                                fill="currentColor"
                                height="1em"
                                preserveAspectRatio="xMidYMid meet"
                                style={
                                  Object {
                                    "color": undefined,
                                    "verticalAlign": "middle",
                                  }
                                }
                                viewBox="0 0 40 40"
                                width="1em">
                                <g>
                                  <path
                                    d="m15 27l17.7-17.7 2.3 2.3-20 20-9.3-9.3 2.3-2.3z" />
                                </g>
                              </svg>
                            </IconBase>
                          </MdDone>
                        </i>
                      </FontIcon>
                    </button>
                  </Button>
                </ButtonTooltiped>
              </ButtonTooltipedInked>
            </div>
            <div
              className="Item-center">
              <span
                className="Item-name">
                Task
              </span>
            </div>
            <div
              className="Item-right">
              <ButtonTooltipedInked
                icon={true}
                inkTransitionEnterTimeout={450}
                inkTransitionLeaveTimeout={300}
                inkTransitionOverlap={150}>
                <ButtonTooltiped
                  icon={true}
                  ink={
                    <InkContainer
                      className={undefined}
                      disabledInteractions={undefined}
                      inkClassName={undefined}
                      inkStyle={undefined}
                      style={undefined}
                      transitionEnterTimeout={450}
                      transitionLeaveTimeout={300}
                      transitionOverlap={150}
                      waitForInkTransition={undefined} />
                  }
                  tooltipDelay={0}
                  tooltipPosition="bottom"
                  tooltipTransitionEnterTimeout={150}
                  tooltipTransitionLeaveTimeout={150}>
                  <Button
                    fixedPosition="br"
                    icon={true}
                    iconBefore={true}
                    ink={
                      <InkContainer
                        className={undefined}
                        disabledInteractions={undefined}
                        inkClassName={undefined}
                        inkStyle={undefined}
                        style={undefined}
                        transitionEnterTimeout={450}
                        transitionLeaveTimeout={300}
                        transitionOverlap={150}
                        waitForInkTransition={undefined} />
                    }
                    type="button">
                    <button
                      className="md-inline-block md-btn md-btn--icon md-pointer--hover"
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseOver={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchStart={[Function]}
                      type="button">
                      <InkContainer
                        transitionEnterTimeout={450}
                        transitionLeaveTimeout={300}
                        transitionOverlap={150}>
                        <ReactTransitionGroup
                          childFactory={[Function]}
                          className="md-ink-container"
                          component="div">
                          <div
                            className="md-ink-container" />
                        </ReactTransitionGroup>
                      </InkContainer>
                      <FontIcon
                        iconClassName="material-icons">
                        <i
                          className="md-icon material-icons">
                          <MdCreate>
                            <IconBase
                              viewBox="0 0 40 40">
                              <svg
                                fill="currentColor"
                                height="1em"
                                preserveAspectRatio="xMidYMid meet"
                                style={
                                  Object {
                                    "color": undefined,
                                    "verticalAlign": "middle",
                                  }
                                }
                                viewBox="0 0 40 40"
                                width="1em">
                                <g>
                                  <path
                                    d="m34.5 11.7l-3 3.1-6.3-6.3 3.1-3q0.5-0.5 1.2-0.5t1.1 0.5l3.9 3.9q0.5 0.4 0.5 1.1t-0.5 1.2z m-29.5 17.1l18.4-18.5 6.3 6.3-18.4 18.4h-6.3v-6.2z" />
                                </g>
                              </svg>
                            </IconBase>
                          </MdCreate>
                        </i>
                      </FontIcon>
                    </button>
                  </Button>
                </ButtonTooltiped>
              </ButtonTooltipedInked>
              <ButtonTooltipedInked
                icon={true}
                inkTransitionEnterTimeout={450}
                inkTransitionLeaveTimeout={300}
                inkTransitionOverlap={150}>
                <ButtonTooltiped
                  icon={true}
                  ink={
                    <InkContainer
                      className={undefined}
                      disabledInteractions={undefined}
                      inkClassName={undefined}
                      inkStyle={undefined}
                      style={undefined}
                      transitionEnterTimeout={450}
                      transitionLeaveTimeout={300}
                      transitionOverlap={150}
                      waitForInkTransition={undefined} />
                  }
                  tooltipDelay={0}
                  tooltipPosition="bottom"
                  tooltipTransitionEnterTimeout={150}
                  tooltipTransitionLeaveTimeout={150}>
                  <Button
                    fixedPosition="br"
                    icon={true}
                    iconBefore={true}
                    ink={
                      <InkContainer
                        className={undefined}
                        disabledInteractions={undefined}
                        inkClassName={undefined}
                        inkStyle={undefined}
                        style={undefined}
                        transitionEnterTimeout={450}
                        transitionLeaveTimeout={300}
                        transitionOverlap={150}
                        waitForInkTransition={undefined} />
                    }
                    type="button">
                    <button
                      className="md-inline-block md-btn md-btn--icon md-pointer--hover"
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseOver={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchStart={[Function]}
                      type="button">
                      <InkContainer
                        transitionEnterTimeout={450}
                        transitionLeaveTimeout={300}
                        transitionOverlap={150}>
                        <ReactTransitionGroup
                          childFactory={[Function]}
                          className="md-ink-container"
                          component="div">
                          <div
                            className="md-ink-container" />
                        </ReactTransitionGroup>
                      </InkContainer>
                      <FontIcon
                        iconClassName="material-icons">
                        <i
                          className="md-icon material-icons">
                          <MdMoreVert>
                            <IconBase
                              viewBox="0 0 40 40">
                              <svg
                                fill="currentColor"
                                height="1em"
                                preserveAspectRatio="xMidYMid meet"
                                style={
                                  Object {
                                    "color": undefined,
                                    "verticalAlign": "middle",
                                  }
                                }
                                viewBox="0 0 40 40"
                                width="1em">
                                <g>
                                  <path
                                    d="m20 26.6c1.8 0 3.4 1.6 3.4 3.4s-1.6 3.4-3.4 3.4-3.4-1.6-3.4-3.4 1.6-3.4 3.4-3.4z m0-10c1.8 0 3.4 1.6 3.4 3.4s-1.6 3.4-3.4 3.4-3.4-1.6-3.4-3.4 1.6-3.4 3.4-3.4z m0-3.2c-1.8 0-3.4-1.6-3.4-3.4s1.6-3.4 3.4-3.4 3.4 1.6 3.4 3.4-1.6 3.4-3.4 3.4z" />
                                </g>
                              </svg>
                            </IconBase>
                          </MdMoreVert>
                        </i>
                      </FontIcon>
                    </button>
                  </Button>
                </ButtonTooltiped>
              </ButtonTooltipedInked>
            </div>
          </div>
        </AccessibleFakeButton>
      </AccessibleFakeButtonInked>
    </Component>
    <Divider>
      <hr
        className="md-divider" />
    </Divider>
  </div>
</Component>
`;
